02-express-tutorial/app.js

```diff
@@ -1 +1,41 @@
-console.log('Express Tutorial')
+const express = require('express')
+const { products } = require("./data");
+
+const app = express();
+
+const port = 3000;
+
+app.use(express.static("./public"))
+
+app.get('/api/v1/test', (req, res) => {
+    res.json({ message: 'It worked!' })
+})
+
+app.get('/api/v1/products', (req, res) => {
+    res.json(products)
+})
+
+app.get('/api/v1/products/:productID', (req, res) => {
+    const id = parseInt(req.params.productID)
+    const findId = products.find((item) => item.id === id)
+    if (!findId) {
+        res.status(404).send('Product not found')
+    } else {
+        res.json(findId)
+    }
+})
+
+app.get('/api/v1/query', (req, res) => {
+    const search = req.query.search
+    const cost = req.query.cost
```

Comment:
If you want to use the cost parameter to filter the products that have the price less than the cost, the logic could be like this:
`filteredProducts = filteredProducts.filter((item) => {
            return item.price < Number(cost);
        });`
You have to check if the cost parameter is defined.